### ISSUE TITLE
Error in getDependencies(..., self=FALSE) for components of multvariate nodes

### ISSUE THREAD

**Issue by danielturek** [nimble-dev/nimble#341]
`model$getDependencies(...)` doesn't respect the `self=FALSE` option, when getting dependencies of a multivariate node, or those of a scalar component of a multivariate node.

This cannot be remedied using `returnScalarComponents = TRUE`, as I thought used to work.

See examples below, which also shows "what I would expect" to see.

```
library(nimble)
code <- nimbleCode({
    x[1:4] ~ dmnorm(mu[1:4], C[1:4,1:4])
    y ~ dexp(x[1])
})
constants <- list(mu = rep(0,4), C = diag(4))
Rmodel <- nimbleModel(code, constants)

> Rmodel$getDependencies('x[1:4]')
##[1] "x[1:4]" "y"     

> Rmodel$getDependencies('x[1:4]', self=FALSE)
##[1] "x[1:4]" "y"      ------ I would expect: "y"

> Rmodel$getDependencies('x[1:4]', returnScalarComponents = TRUE)
##[1] "x[1]" "x[2]" "x[3]" "x[4]" "y"   

> Rmodel$getDependencies('x[1:4]', self=FALSE, returnScalarComponents = TRUE)
## [1] "x[1]" "x[2]" "x[3]" "x[4]" "y"   ------ I would expect: "y"

> Rmodel$getDependencies('x[1]')
## [1] "x[1:4]" "y"     

> Rmodel$getDependencies('x[1]', self=FALSE)
## [1] "x[1:4]" "y"     ------ I would expect: "y"

> Rmodel$getDependencies('x[1]', returnScalarComponents = TRUE)
## [1] "x[1]" "x[2]" "x[3]" "x[4]" "y"   

> Rmodel$getDependencies('x[1]', self=FALSE, returnScalarComponents = TRUE)
## [1] "x[1]" "x[2]" "x[3]" "x[4]" "y"    ------ I would expect: "y"
```

**Comment by perrydv** [comment]
I'm guessing the `omit` argument also doesn't work (and is never used internally but should be supported).

The `self=FALSE` bug arises because the `nodeIDs` used to filter out `self` previously represented graphIDs but now represent vertexIDs, so they won't correctly filter out the graphIDs in depIDs at [this point](https://github.com/nimble-dev/nimble/blob/e8cfb1240000fbfa1f4c8d70bdbf2c4c05f62ad1/packages/nimble/R/BUGS_model.R#L693).  This looks like it will be a simple fix.

**Comment by danielturek** [comment]
Great.  I'll stay tuned.

**Comment by perrydv** [comment]
fixed by pull request #346
